core/descriptors/src/org/jetbrains/kotlin/descriptors/FunctionDescriptor.java

```diff
@@ -79,6 +79,8 @@
 
     boolean isSuspend();
 
+    boolean isGenerated();
```

Comment:
> so descriptor retrieved from compiled code are equivalent to descriptors before compilation.

Can you explain that in a bit more detail?

Also, @goodwinnk, wouldn't this information be good to have on the `FunctionDescriptor` interface to prevent future developers from making the same mistake?